packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,666 @@
+                                        flutter::EncodableValue(track.index));
+        trackSelection.insert_or_assign(
+            flutter::EncodableValue("language"),
+            flutter::EncodableValue(track.language_code));
+        LOG_INFO("[PlusPlayer] subtitle track info[%d]: language[%s]",
+                 track.index, track.language_code.c_str());
+
+        trackSelections.push_back(flutter::EncodableValue(trackSelection));
+      }
+    }
+  }
+
+  return trackSelections;
+}
+
+bool PlusPlayer::SetTrackSelection(int32_t track_id, int32_t track_type) {
+  LOG_INFO("[PlusPlayer] track_id: %d,track_type: %d", track_id, track_type);
+
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Player not created.");
+    return false;
+  }
+
+  plusplayer::State state = player_->GetState();
+  if (state == plusplayer::State::kNone || state == plusplayer::State::kIdle) {
+    LOG_ERROR("[PlusPlayer] Player is in invalid state.");
+    return false;
+  }
+
+  plusplayer::TrackType type;
+  switch (track_type) {
+    case 1:
+      type = plusplayer::TrackType::kTrackTypeAudio;
+      break;
+    case 2:
+      type = plusplayer::TrackType::kTrackTypeVideo;
+      break;
+    case 3:
+      type = plusplayer::TrackType::kTrackTypeSubtitle;
+      break;
+    default:
+      LOG_ERROR("[PlusPlayer] Invalid track type: %d", track_type);
+      return {};
+  }
+  if (!player_->SelectTrack(type, track_id)) {
+    LOG_ERROR("[PlusPlayer] Player select track failed.");
+    return false;
+  }
+  return true;
+}
+
+bool PlusPlayer::SetDrm(const std::string &uri, int drm_type,
+                        const std::string &license_server_url) {
+  drm_manager_ = std::make_unique<DrmManager>();
+  if (!drm_manager_->CreateDrmSession(drm_type, true)) {
+    LOG_ERROR("[PlusPlayer] Failed to create drm session.");
+    return false;
+  }
+
+  int drm_handle = 0;
+  if (!drm_manager_->GetDrmHandle(&drm_handle)) {
+    LOG_ERROR("[PlusPlayer] Failed to get drm handle.");
+    return false;
+  }
+
+  plusplayer::drm::Type type;
+  switch (drm_type) {
+    case DrmManager::DrmType::DRM_TYPE_PLAYREADAY:
+      type = plusplayer::drm::Type::kPlayready;
+      break;
+    case DrmManager::DrmType::DRM_TYPE_WIDEVINECDM:
+      type = plusplayer::drm::Type::kWidevineCdm;
+      break;
+    default:
+      type = plusplayer::drm::Type::kNone;
+      break;
+  }
+
+  plusplayer::drm::Property property;
+  property.handle = drm_handle;
+  property.type = type;
+  property.license_acquired_cb =
+      reinterpret_cast<plusplayer::drm::LicenseAcquiredCb>(OnLicenseAcquired);
+  property.license_acquired_userdata =
+      reinterpret_cast<plusplayer::drm::UserData>(this);
+  property.external_decryption = false;
+  player_->SetDrm(property);
+
+  if (license_server_url.empty()) {
+    bool success = drm_manager_->SetChallenge(
+        uri, [this](const void *challenge, unsigned long challenge_len,
+                    void **response, unsigned long *response_len) {
```

Comment:
```suggestion
        uri, [this](const std::string &challenge) -> std::vector<uint8_t> {
```
This should be cleaner and readable. (Please refer to the videohole plugin implementation.)